src/main/java/org/candlepin/dto/shim/ProductDTOTranslator.java

```diff
@@ -85,22 +91,65 @@ public ProductData populate(ModelTranslator modelTranslator, ProductDTO source,
         // We manually set this to false since it is not included in the exported data.
         dest.setLocked(false);
 
+        if (modelTranslator != null) {
+            dest.setProductContent(translateProductContent(modelTranslator, source));
+            dest.setBranding(translateBranding(modelTranslator, source));
+            dest.setProvidedProducts(translateProvidedProducts(modelTranslator, source));
+        }
+        else {
+            dest.setProductContent(new HashSet<>());
+            dest.setBranding(new HashSet<>());
+            dest.setProvidedProducts(new HashSet<>());
+        }
+
+        return dest;
+    }
+
+    private Set<ProductContentData> translateProductContent(
+        ModelTranslator modelTranslator, ProductDTO source) {
         Collection<ProductDTO.ProductContentDTO> productContentDTOs = source.getProductContent();
-        dest.setProductContent(Collections.emptyList());
-
-        if (modelTranslator != null && productContentDTOs != null) {
-            ObjectTranslator<ContentDTO, ContentData> contentDTOTranslator = modelTranslator
-                .findTranslatorByClass(ContentDTO.class, ContentData.class);
-
-            for (ProductDTO.ProductContentDTO pcdto : productContentDTOs) {
-                if (pcdto != null && pcdto.getContent() != null) {
-                    ContentData contentData = contentDTOTranslator.translate(modelTranslator,
-                        pcdto.getContent());
-                    dest.addContent(contentData, pcdto.isEnabled());
-                }
+        if (productContentDTOs == null) {
+            return new HashSet<>();
+        }
+
+        ObjectTranslator<ContentDTO, ContentData> contentDTOTranslator = modelTranslator
+            .findTranslatorByClass(ContentDTO.class, ContentData.class);
+        Set<ProductContentData> productContent = new HashSet<>();
+        for (ProductDTO.ProductContentDTO pcdto : productContentDTOs) {
+            if (pcdto != null && pcdto.getContent() != null) {
+                ContentData content = contentDTOTranslator.translate(modelTranslator, pcdto.getContent());
+                productContent.add(new ProductContentData(content, pcdto.isEnabled()));
             }
         }
 
-        return dest;
+        return productContent;
+    }
+    private Set<Branding> translateBranding(ModelTranslator modelTranslator, ProductDTO source) {
+        Set<BrandingDTO> branding = source.getBranding();
+        if (branding == null) {
+            return new HashSet<>();
+        }
+
+        ObjectTranslator<BrandingDTO, Branding> translator = modelTranslator
+            .findTranslatorByClass(BrandingDTO.class, Branding.class);
+
+        return branding.stream()
+            .filter(Objects::nonNull)
+            .map(p -> translator.translate(modelTranslator, p))
```

Comment:
Fetching the translator above saves us from the repeated resolution, but we still could use the stream mapper function to streamline the code here.

Reply:
Done.